diff --git a/legal.module b/legal.module
index 657f01e..415cd7b 100644
--- a/legal.module
+++ b/legal.module
@@ -355,6 +355,7 @@ function legal_form_user_profile_form_alter(&$form, $form_state) {
 
   // Enable account owner to accept.
   if ($account->uid == $user->uid && $accepted != TRUE) {
+    drupal_set_message(t('The Terms & Conditions have been updated. Please review the new Terms & Conditions below.'));
     $form['legal']['legal_accept']['#default_value'] = isset($edit['legal_accept']) ? $edit['legal_accept'] : '';
     $form['legal']['legal_accept']['#required']      = TRUE;
 
@@ -380,6 +381,11 @@ function legal_user_login(&$edit, $account) {
   global $user;
   global $language;
 
+  // don't block the password reset form
+  if (empty($edit['form_id']) || $edit['form_id'] == 'user_pass_reset') {
+    return;
+  }
+
   if ($user->uid == 1) {
     return;
   }
